atomate/vasp/workflows/base/exchange.py

```diff
@@ -0,0 +1,183 @@
+                relax to a low-spin configuration during a DFT electronic
+                configuration.
+            db_file (string): Path to database file.
+            name (string): Name of workflow.
+
+        """
+
+        self.uuid = str(uuid4())
+        self.wf_meta = {
+            "wf_uuid": self.uuid,
+            "wf_name": self.__class__.__name__,
+            "wf_version": __version__,
+        }
+
+        # Make sure gs is index 0
+        ordered_structures = [
+        s for _, s in sorted(zip(energies, magnetic_structures), reverse=False)]
+        ordered_energies = sorted(energies, reverse=False)
+
+        self.structures = ordered_structures
+        self.energies = ordered_energies
+
+        # Check for magmoms
+        for s in magnetic_structures:
+            try:
+                magmoms = s.site_properties["magmom"]
+            except:
+                raise RuntimeError(
+                    "All structures must have 'magmom' site \
+                    property."
+                )
+
+
+    def get_wf(self, num_orderings_hard_limit=16, c=None):
+        """Retrieve Fireworks workflow.
+
+        c is a dictionary that can contain:
+        * heisenberg_settings: 
+            cutoff (float): Starting point for nearest neighbor search.
+            tol (float): Tolerance for equivalent NN bonds.
+            average (bool): Compute only <J>.
+        * mc_settings:
+            mc_box_size (float): MC simulation box size in nm.
+            equil_timesteps (int): Number of MC equilibration moves.
+            mc_timesteps (int): Number of MC moves for averaging.
+        * DB_FILE:
+            path to db.json.
+
+        Args:
+            num_orderings_hard_limit (int): will make sure total number of
+                magnetic orderings does not exceed this number even if there
+                are extra orderings of equivalent symmetry
+            c (dict): additional config dict described above
+
+        Returns: 
+            wf (Workflow): Static magnetic orderings + Heisenberg Model +
+                Vampire Monte Carlo.
+
+        TODO:
+            * Add static SCAN option (only optimization is available)
+
+        """
+
+        c = c or {"DB_FILE": DB_FILE}
+        
+        if "DB_FILE" not in c:
+            c["DB_FILE"] = DB_FILE
+
+        heisenberg_settings = {"cutoff": 3.0, "tol": 0.04, "average": True}
+        heisenberg_settings.update(c.get("heisenberg_settings", {}))
+        c["heisenberg_settings"] = heisenberg_settings
+
+        fws = []
+
+        heisenberg_model_fw = HeisenbergModelFW(
+            wf_uuid=self.uuid,
+            parent_structure=self.structures[0],
+            db_file=c["DB_FILE"],
+            heisenberg_settings=c["heisenberg_settings"],
+            parents=None,
+            structures=self.structures,
+            energies=self.energies,
+        )
+
+        # Vampire Monte Carlo
+        mc_settings = {
```

Comment:
These are the same as the `VampireCaller` defaults. Currently this results in duplication of settings. Ideally if you change the default VampireCaller defaults to make them better, you'd then have to change them in atomate too. If you follow the above by making this an optional argument and then don't set it, we can use the Pymatgen defaults for free.

Reply:
Sounds good, I'll do the same with the `HeisenbergMapper` defaults as those are defined in pymatgen as well.